src/app/backend/resource/common/controller.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2017 The Kubernetes Dashboard Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// IsControlledBy checks if the  object has a controllerRef set to the given owner.
+func IsControlledBy(obj v1.Object, owner v1.Object) bool {
+	ref := GetControllerOf(obj)
+	if ref == nil {
+		return false
+	}
+	return ref.UID == owner.GetUID()
+}
+
+// GetControllerOf returns a pointer to a copy of the controllerRef if controllee has a controller.
+func GetControllerOf(controllee v1.Object) *v1.OwnerReference {
```

Comment:
Since `client-go` is stable and has almost all functionalities we tend to avoid using `kubernetes` dependencies directly, as seems to be "cleaner" way. Few weeks ago our repository `vendor` directory had like 60-65 MB, now it's around 20 MB. Sadly, we have to copy some functions from core from time to time. We try to replace them with calls to available API when it becomes possible. At the moment we have 2-3 places where we copied core methods, I think it is not that bad result.